Sources/SwiftCurrent_SwiftUI/Views/WorkflowView.swift

```diff
@@ -0,0 +1,215 @@
+    private var onFinish = [(AnyWorkflow.PassedArgs) -> Void]()
+    private var onAbandon = [() -> Void]()
+    private var passedArgs = AnyWorkflow.PassedArgs.none
+
+    /// Creates a `WorkflowView` that displays a `FlowRepresentable` when presented.
+    public init(isPresented: Binding<Bool>) where Args == Never {
+        _isPresented = isPresented
+    }
+
+    /// Creates a `WorkflowView` that displays a `FlowRepresentable` when presented.
+    public init(isPresented: Binding<Bool>, args: Args) {
+        _isPresented = isPresented
+        if let args = args as? AnyWorkflow.PassedArgs {
+            passedArgs = args
+        } else {
+            passedArgs = .args(args)
+        }
+    }
+
+    private init(isPresented: Binding<Bool>,
+                 workflow: AnyWorkflow?,
+                 onFinish: [(AnyWorkflow.PassedArgs) -> Void],
+                 onAbandon: [() -> Void],
+                 passedArgs: AnyWorkflow.PassedArgs) {
+        _isPresented = isPresented
+        self.workflow = workflow
+        self.onFinish = onFinish
+        self.onAbandon = onAbandon
+        self.passedArgs = passedArgs
+    }
+
+    public var body: some View {
+        if isPresented {
+            VStack {
+                model.body
+            }
+            .onAppear {
+                guard !didLoad else { return }
+                didLoad = true
+                model.isPresented = $isPresented
+                model.onAbandon = onAbandon
+                workflow?.launch(withOrchestrationResponder: model,
+                                 passedArgs: passedArgs,
+                                 launchStyle: .new) { passedArgs in
+                    onFinish.forEach { $0(passedArgs) }
+                }
+            }
+            .onDisappear {
+                model.body = AnyView(EmptyView())
+            }
+            .onReceive(inspection.notice) { inspection.visit(self, $0) } // Needed for ViewInspector
+        }
+    }
+
+    /// Adds an action to perform when this `Workflow` has finished.
+    public func onFinish(closure: @escaping (AnyWorkflow.PassedArgs) -> Void) -> Self {
+        var onFinish = self.onFinish
+        onFinish.append(closure)
+        return WorkflowView(isPresented: $isPresented,
+                            workflow: workflow,
+                            onFinish: onFinish,
+                            onAbandon: onAbandon,
+                            passedArgs: passedArgs)
+    }
+
+    /// Adds an action to perform when this `Workflow` has abandoned.
+    public func onAbandon(closure: @escaping () -> Void) -> Self {
+        var onAbandon = self.onAbandon
+        onAbandon.append(closure)
+        return WorkflowView(isPresented: $isPresented,
+                            workflow: workflow,
+                            onFinish: onFinish,
+                            onAbandon: onAbandon,
+                            passedArgs: passedArgs)
+    }
+}
+
+// MARK: thenProceeds
```

Comment:
We can leave this if it is helpful but otherwise I think this was accidentally left in.
```suggestion
```

Reply:
Removed